### ISSUE TITLE
feature request: progress for unknown number of steps

### ISSUE THREAD

**Issue by tpapp** [timholy/ProgressMeter.jl#101]
Suppose I am reading an *ex ante* unknown number of records from a large file. It would be nice to do something like

```julia
prog = ProgressOnline("reading records")
while !eof(io)
    line = readline(io)
    ProgressMeter.update!(prog) # would just increment a counter, no second arg
    ... # process line
end
ProgressMeter.update!(prog, nothing) # done
```

while would display something like
```julia
reading records: 42000
```

**Comment by ChrisRackauckas** [comment]
This would be solved by https://github.com/timholy/ProgressMeter.jl/issues/102 since that logging information lets you give the update percentage.

**Comment by tpapp** [comment]
#102 is nice, but this issue is about the case where I **don't know** the percentage, since I don't know the total.

**Comment by zsunberg** [comment]
Just out of curiosity, what is the advantage of this over just printing with a carriage return at the beginning?

```
i = 0
while !eof(io)
    line = readline(io)
    print("\rProcessed $i lines")
    ... # process line
end
println()
```

**Comment by saschatimme** [comment]
@zsunberg  The implementation in #115  has a couple of advantages over your suggested solutions. 
It also automatically keeps track of the passed time and prints it nicely.
Also you can also easily specify additional values to be printed (with the `showvalues` keyword). Sure, you could all do this by your own, but it is also nice if everybody doesn't have to reinvent the wheel :)

Another major benefit of the implementation in #115 is that it updates the current information only every, let's say 0.1 seconds. Depending on the task this reduces the overhead of the progress bar immensely since IO operations are usually quite expensive.

**Comment by zsunberg** [comment]
Great, thanks for the explanation

**Comment by martinholters** [comment]
Closed by #115.
